app.rb

```diff
@@ -0,0 +1,76 @@
+require_relative 'book'
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'rental'
+
+def list_all_books(books)
+  if books.empty?
+    puts 'There is no book!'
+  else
+    books.each { |book| puts "Title: '#{book.title}', Author: #{book.author}" }
+  end
+end
+
+def list_all_people(people)
+  if people.empty?
+    puts 'there is no person'
+  else
+    people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+  end
+end
+
+def create_person(person)
+  puts 'do you want to create a student(1) or a teacher (2):'
+  selected_person = gets.chomp.to_i
+  print 'age:'
+  age = gets.chomp.to_i
+  print 'name:'
+  name = gets.chomp.to_s
+  case selected_person
+  when 1
+    print 'Has parent permission? [Y/N]: '
+    provided_permission = gets.chomp.capitalize
+    student_permission = true if provided_permission == 'Y'
+    student_permission = false if provided_permission == 'N'
+    person.push(Student.new(nil, age, name, parent_permission: student_permission))
+  when 2
+    print 'Specialization: '
+    specialization = gets.chomp
+    person.push(Teacher.new(specialization, age, name))
```

Comment:
- [ ] there is an error when trying to create a teacher, this is because you spelt your Teacher class as `Teachear` also you did not pass name as a parameter in your constructor, kindly fix that
![Screenshot from 2022-08-25 17-32-11](https://user-images.githubusercontent.com/66097371/186720975-6949ea9d-14d1-4169-ae3f-83844c7728f4.png)

Reply:
thank you @VICTOR-CHUKWU